phys/module_ltng_pe.F

```diff
@@ -0,0 +1,323 @@
+                                                      light_ne, &
+                                                      light_neu
+
+      REAL,INTENT(IN):: dx,dy
+      REAL, DIMENSION( ims:ime, kms:kme, jms:jme ),                  &
+         INTENT(IN ) ::  w
+   REAL, DIMENSION( ims:ime , kms:kme , jms:jme ),                    &
+         INTENT(IN) ::                                         th_phy
+!
+
+!
+   REAL, DIMENSION( ims:ime , kms:kme , jms:jme ),                    &
+         INTENT(IN   ) ::                                             &
+                                                                rho_phy, &
+                                                                dz8w, &
+                                                              pi_phy, &
+                                                                   p_phy
+
+!     REAL, INTENT(INOUT),  DIMENSION(ims:ime,jms:jme)::      &
+!    &                      prec_ice,wqg_m15
+      REAL, DIMENSION(ims:ime,jms:jme)::      &
+     &                      LPI,LPOS,LNEG,LNEU,plpi,nlpi,neulpi
+!     REAL, DIMENSION(ims:ime,jms:jme)::      &
+!    &                      LPI,LPOS,LNEG,LNEU,plpi,nlpi,neulpi,prec_ice,wqg_m15
+
+
+
+
+      REAL, DIMENSION(kms:kme)::    tempk,rh,qtot,qitot
+      REAL, DIMENSION(kms:kme):: p1d,rho1d,qti1d
+      REAL, DIMENSION(kms:kme):: temp,qc1d,ql1d,qi1d,qs1d,qg1d,lpi1d
+      REAL, DIMENSION(0:kme):: w1d,height
+      REAL, DIMENSION(kms:kme):: e1d,height_t,w1d_t
+      REAL z_full,qrs,teten,RELHUM,LOC,Td_850,Td_700,PC_DWPT
+      INTEGER level
+      REAL :: dt,dxdy,t_base,t_top
+      INTEGER I_COLLAPSE
+      LOGICAL LOOK_T
+      real t_pos,t_neg,t_neu
+      INTEGER I_START,I_END,J_START,J_END
+
+
+  INTEGER ::               i,j,k
+!-------------------------------------------------------------------
+! !   print*,'ims,ime,jms,jme,kms,kme = ',ims,ime,jms,jme,kms,kme 
+!     print*,'its,ite,jts,jte,kts,kte = ',its,ite,jts,jte,kts,kte 
+!     parameter(t_pos=0.12,t_neg=0.06,t_neu=0.03) 
+!     note: t_pos is modified based on the height of the top of the cloud
+!     I believe that the value of 60000 m/s might be a typical speed
+!     See: https\://en.wikipedia.org/wiki/Lightning
+      t_pos=0.12
+      t_neg=0.06
+      t_neu=0.03
+      dxdy=dx*dy
+!     print*,'kte = ',kte
+!     return (1)
+      DO j = jts,jte
+      DO i = its,ite
+        z_full=0.
+        height(0)=z_full
+        w1d(0)=w(i,1,j)
+      DO k = kts,kte
+          if (k.lt.kte)then
+           w1d(k)=w(i,k+1,j)
+          else
+           w1d(k)=0.
+          end if
+          temp(k) = th_phy(i,k,j)*pi_phy(i,k,j)-273.16
+          tempk(k) = th_phy(i,k,j)*pi_phy(i,k,j)
+          p1d(k)=p_phy(i,k,j)
+          rho1d(k)=rho_phy(i,k,j)
+          z_full=z_full+dz8w(i,k,j)
+          height(k)=z_full
+          qc1d(k)=qc(i,k,j)
+          ql1d(k)=qc(i,k,j)+qr(i,k,j)
+          qi1d(k)=qi(i,k,j)
+!         qti1d(k)=qi(i,k,j)+qs(i,k,j)+qg(i,k,j)+qh(i,k,j)
```

Comment:
@BarryHLynn 
Barry,
The lines that are commented out all have hail. Can they be organized, or a comment provided for users?